.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,48 @@
+name: Docker-Build Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  # TODO: disable until caching system is implemented
+  # pull_request:
+  #   types: [opened, reopened, synchronize, ready_for_review, review_requested]
+  #   branches:
+  #     - main
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Docker Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.9
+
+      - name: Install nox
+        run: |
+          pip install nox
+          # TODO: remove this as a sessiond dependency
```

Comment:
This (`pip install pyyaml`) needed to be added because the `noxfile.py` has a session that imports and uses `yaml` within the body of the function rather than `session.run(...)`, so `session.install('pyyaml')` doesn't solve it.

It's simple enough to install here, but in principle, we should probably refactor the `build_algorithm` session to use the `yaml` module within the session's environment.